src/Microsoft.OData.Client/BaseEntityType.cs

```diff
@@ -13,5 +13,9 @@ public class BaseEntityType
     {
         /// <summary>DataServiceContext for query provider</summary>
         protected internal DataServiceContext Context { get; set; }
+        /// <summary>
+        /// Stream descriptor for the entity  containing entity stream links
+        /// </summary>
+        protected internal StreamDescriptor StreamDescriptor { get; set; }

```

Comment:
I didn't find the place to set this Descriptor?

Reply:
It is set inside the materializer, when certain conditions are met, in this case an entity that has a stream and  the context is in NoTracking mode. In normal tracking mode the descriptor is available in the entity tracker.